--- /home/marek/qemu-orig/hw/acpi/aml-build.c	2026-03-20 18:27:00.654366544 +0100
+++ /home/marek/qemu/hw/acpi/aml-build.c	2026-03-19 22:38:59.966857676 +0100
@@ -1722,11 +1722,11 @@
     build_append_int_noprefix(array, 0, 4); /* Length */
     build_append_int_noprefix(array, desc->rev, 1); /* Revision */
     build_append_int_noprefix(array, 0, 1); /* Checksum */
-    build_append_padded_str(array, desc->oem_id, 6, '\0'); /* OEMID */
+    build_append_padded_str(array, ACPI_BUILD_APPNAME6, 6, '\0'); /* OEMID */
     /* OEM Table ID */
-    build_append_padded_str(array, desc->oem_table_id, 8, '\0');
+    build_append_padded_str(array, "A M I   ", 8, '\0');
     build_append_int_noprefix(array, 1, 4); /* OEM Revision */
-    g_array_append_vals(array, ACPI_BUILD_APPNAME8, 4); /* Creator ID */
+    g_array_append_vals(array, "A M I", 4); /* Creator ID */
     build_append_int_noprefix(array, 1, 4); /* Creator Revision */
 }
 
@@ -2346,7 +2346,9 @@
     build_append_gas_from_struct(tbl, &f->gpe0_blk); /* X_GPE0_BLK */
     build_append_gas(tbl, AML_AS_SYSTEM_MEMORY, 0 , 0, 0, 0); /* X_GPE1_BLK */
 
-    if (f->rev <= 4) {
+    if (f->rev <= 6) {
+		build_append_gas_from_struct(tbl, &f->sleep_ctl);
+		build_append_gas_from_struct(tbl, &f->sleep_sts);
         goto done;
     }
 
@@ -2360,7 +2362,7 @@
     }
 
     /* Hypervisor Vendor Identity */
-    build_append_padded_str(tbl, "QEMU", 8, '\0');
+    build_append_padded_str(tbl, "Intel", 8, '\0');
 
     /* TODO: extra fields need to be added to support revisions above rev6 */
     assert(f->rev == 6);
diff -ru /home/marek/qemu-orig/hw/smbios/smbios.c /home/marek/qemu/hw/smbios/smbios.c
--- /home/marek/qemu-orig/hw/smbios/smbios.c	2026-03-20 18:27:00.735721734 +0100
+++ /home/marek/qemu/hw/smbios/smbios.c	2026-03-19 21:36:37.063463308 +0100
@@ -708,7 +708,7 @@
     t->max_speed = cpu_to_le16(type4.max_speed);
     t->current_speed = cpu_to_le16(type4.current_speed);
     t->status = 0x41; /* Socket populated, CPU enabled */
-    t->processor_upgrade = 0x01; /* Other */
+    t->processor_upgrade = 0x36; /* AM4 */
     t->l1_cache_handle = cpu_to_le16(0xFFFF); /* N/A */
     t->l2_cache_handle = cpu_to_le16(0xFFFF); /* N/A */
     t->l3_cache_handle = cpu_to_le16(0xFFFF); /* N/A */
--- /home/marek/qemu-orig/include/hw/acpi/aml-build.h	2026-03-20 18:27:00.761259716 +0100
+++ /home/marek/qemu/include/hw/acpi/aml-build.h	2026-03-19 22:38:59.966857676 +0100
@@ -4,8 +4,8 @@
 #include "hw/acpi/acpi-defs.h"
 #include "hw/acpi/bios-linker-loader.h"
 
-#define ACPI_BUILD_APPNAME6 "BOCHS "
-#define ACPI_BUILD_APPNAME8 "BXPC    "
+#define ACPI_BUILD_APPNAME6 "ALASKA"
+#define ACPI_BUILD_APPNAME8 "PC8086    "
 
 #define ACPI_BUILD_TABLE_FILE "etc/acpi/tables"
 #define ACPI_BUILD_RSDP_FILE "etc/acpi/rsdp"
--- /home/marek/qemu-orig/include/hw/pci/pci.h	2026-03-20 18:27:00.778262034 +0100
+++ /home/marek/qemu/include/hw/pci/pci.h	2026-03-19 14:31:14.240460558 +0100
@@ -100,7 +100,7 @@
  */
 #define PCI_DEVICE_ID_VIRTIO_10_BASE     0x1040
 
-#define PCI_VENDOR_ID_REDHAT             0x1b36
+#define PCI_VENDOR_ID_REDHAT             0x8086
 #define PCI_DEVICE_ID_REDHAT_BRIDGE      0x0001
 #define PCI_DEVICE_ID_REDHAT_SERIAL      0x0002
 #define PCI_DEVICE_ID_REDHAT_SERIAL2     0x0003
